test/suites/zombie-tanssi-relay-eth-bridge/test-consensus-bridge.ts

```diff
@@ -0,0 +1,156 @@
+                shell: true,
+                detached: true,
+            });
+            ethereumNodeChildProcess.stderr.setEncoding("utf-8");
+            ethereumNodeChildProcess.stderr.on("data", (chunk) => console.log(chunk));
+
+            await execCommand("./scripts/bridge/generate-beefy-checkpoint.sh", {
+                env: {
+                    RELAYCHAIN_ENDPOINT: "ws://127.0.0.1:9947",
+                    ...process.env,
+                },
+            });
+
+            // Waiting till ethreum node produces one block
+            console.log("Waiting some time for ethereum node to produce block, before we deploy contract");
+            await sleep(20000);
+
+            await execCommand("./scripts/bridge/deploy-ethereum-contracts.sh");
+
+            console.log("Contracts deployed");
+
+            const contractInfoData = JSON.parse(
+                <string>(await execCommand("./scripts/bridge/generate-contract-info.sh")).stdout
+            );
+            console.log("BeefyClient contract address is:", contractInfoData.contracts.BeefyClient.address);
+            beefyClientDetails = contractInfoData.contracts.BeefyClient;
+
+            const initialBeaconUpdate = JSON.parse(
+                <string>(
+                    await execCommand("./scripts/bridge/setup-relayer.sh", {
+                        env: {
+                            RELAYCHAIN_ENDPOINT: "ws://127.0.0.1:9947",
+                            ...process.env,
+                        },
+                    })
+                ).stdout
+            );
+
+            // We need to read initial checkpoint data and address of gateway contract to setup the ethereum client
+            // Once that is done, we can start the relayer
+            await signAndSendAndInclude(
+                relayApi.tx.sudo.sudo(relayApi.tx.ethereumBeaconClient.forceCheckpoint(initialBeaconUpdate)),
+                alice
+            );
+
+            relayerChildProcess = spawn("./scripts/bridge/start-relayer.sh", {
+                shell: true,
+                detached: true,
+                env: {
+                    RELAYCHAIN_ENDPOINT: "ws://127.0.0.1:9947",
+                    ...process.env,
+                },
+            });
+            relayerChildProcess.stderr.setEncoding("utf-8");
+            relayerChildProcess.stderr.on("data", (chunk) => console.log(chunk));
+        }, 12000000);
+
+        it({
+            id: "T01",
+            title: "Ethereum Blocks are being recognized on tanssi-relay",
+            test: async function () {
+                await sleep(10000);
+                const firstFinalizedBlockRoot = (
+                    await relayApi.query.ethereumBeaconClient.latestFinalizedBlockRoot()
+                ).toJSON();
+                expect(firstFinalizedBlockRoot).to.not.equal(
+                    "0x0000000000000000000000000000000000000000000000000000000000000000"
+                );
+                await sleep(20000);
```

Comment:
we can possibly wait for blocks instead of time here. Approximately every how many blocks is this injected?

Reply:
*fixed*